### DISCUSSION TITLE
RuntimeError: name too long

### DISCUSSION THREAD

**Discussion by skylin008** [micropython/micropython#14214] [STM32 / Pyboard]
I used pyboard and rebuild the micropython firmware , the  micropython  version  is : v1.22.   I usually  used CTRL + D  key  reset the pboard  to debug the python file, any show off error, how to solve this issue, thanks!
>>>
MPY: sync filesystems
MPY: soft reboot
RuntimeError: name too long
RuntimeError: name too long
MicroPython v1.22

**Comment by dpgeorge** [comment]
What code are you running in your `boot.py` and `main.py` files?  There must be some code there that has a very long name.  What files are on your pyboard's filesystem?

**Comment by tstenvold** [comment]
Your files are likely being corrupted due to a file write and then power loss. This results in the filesystem being intact, but the files are corrupted on the FAT partition.

My guess is you are doing a long write to the flash or SD card and then resetting the power during this write.
